packages/integration_test/lib/common.dart

```diff
@@ -121,3 +132,105 @@ class Failure {
     return Failure(failure['methodName'], failure['details']);
   }
 }
+
+/// Integration web tests can execute WebDriver actions such as screenshots.
+///
+/// These test will use [TestStatus] to notify `integration_test` of their
+/// state.
+enum TestStatus {
+  /// Test is waiting for executing WebDriver actions.
+  waitOnWebdriverCommand,
+
+  /// Test executed the previously requested action.
+  webdriverCommandComplete,
+}
+
+/// Types of different WebDriver actions that can be used in web integration
+/// tests.
+///
+/// These actions are either commands that WebDriver can execute or used
+/// for the communication between `integration_test` and the driver test.
+enum WebDriverActionTypes {
+  /// Acknowlegement for the previously sent message.
+  ack,
+
+  /// No further WebDriver Action is necessary.
+  noop,
+
+  /// Asking WebDriver to take a screenshot of the Web page.
+  screenshot,
+}
+
+/// Command for WebDriver to execute.
+///
+/// Only works on Web when tests are run via `flutter driver` command.
+///
+/// See: https://www.w3.org/TR/webdriver/
+class WebDriverAction {
+  /// Type of the [WebDriverAction].
+  ///
+  /// Currently the only action that triggers a WebDriver API command is
+  /// `screenshot`.
+  ///
+  /// There are also `ack` and `noop` actions defined to manage the handshake
+  /// during the communication.
+  final WebDriverActionTypes type;
+
+  /// Used for adding extra values to the actions such as file name for
+  /// `screenshot`.
+  final Map<String, dynamic> values;
+
+  /// Constructor for [WebDriverActionTypes.noop] action.
+  WebDriverAction.noop()
+      : this.type = WebDriverActionTypes.noop,
+        this.values = Map();
+
+  /// Constructor for [WebDriverActionTypes.noop] screenshot.
+  WebDriverAction.screenshot(String screenshot_name)
+      : this.type = WebDriverActionTypes.screenshot,
+        this.values = {'screenshot_name': screenshot_name};
+
+  /// Util method for converting [WebDriverActionTypes] to a map entry.
+  ///
+  /// Used for converting messages to json format.
+  static Map<String, dynamic> typeToMap(WebDriverActionTypes type) => {
+        'web_driver_action': '${type}',
```

Comment:
Are we certain that in all modes `type.toString` will produce the original name of the command? I don't believe the Dart language spec guarantees that. For example, dart2js minifies the names (e.g. "screenshot" can become "aZ").

Reply:
Interesting I usually run the tests in all modes. Didn't had an issue for now. What is the canonical way of getting a name for an enum like WebDriverActionTypes